src/admin/admin.controller.ts

```diff
@@ -1,4 +1,166 @@
+    @Get("/:adminId/transactions/users")
+    // get all transactions of all users
+    async getAllUsersTransactions(
+        @Param("adminId", ParseIntPipe) adminId: number
+    ) {
+        // check admin
+        await this.adminService.getAdminWallet(adminId);
+        
+        // fetch transactions
+        const transactions = await this.transactionService.fetchAllUsersTransactions();
+        return {
+            statusCode: HttpStatus.OK,
+            message: "transactions fetched successfully",
+            body: {
+                transactions
+            }
+        }
+    }
+
+    @ApiOperation({ summary: 'Get One User Transactions' })
+    @ApiResponse({ status: HttpStatus.OK, description: 'transactions fetched successfully', type: [Transaction] })
+    @Get("/:adminId/transactions/users/:userId")
+    // get all transactions of a particular user
+    async getUserTransactions(
+        @Param("adminId", ParseIntPipe) adminId: number,
+        @Param("userId", ParseIntPipe) userId: number
+    ) {
+        // check admin
+        await this.adminService.getAdminWallet(adminId);
+
+        // check user
+        await this.userService.getUserWallet(userId);
+        
+        // fetch transactions
+        const transactions = await this.transactionService.fetchTransactionsOfOneSystemActor(userId);
+        return {
+            statusCode: HttpStatus.OK,
+            message: "transactions fetched successfully",
+            body: {
+                transactions
+            }
+        }
+    }
+
+    @ApiOperation({ summary: 'Get All Admin Providers Transactions' })
+    @ApiResponse({ status: HttpStatus.OK, description: 'transactions fetched successfully', type: [Transaction] })
+    @Get("/:adminId/transactions/providers")
+    // get all transactions between all providers and admins
+    async getAllAdminProvidersTransactions(
+        @Param("adminId", ParseIntPipe) adminId: number
+    ) {
+        // check admin
+        await this.adminService.getAdminWallet(adminId);
+        
+        // fetch transactions
+        const transactions = await this.transactionService.fetchAllAdminProviderTransactions();
+        return {
+            statusCode: HttpStatus.OK,
+            message: "transactions fetched successfully",
+            body: {
+                transactions
+            }
+        }
+    }
+
+    @ApiOperation({ summary: 'Get One Provider Transactions' })
+    @ApiResponse({ status: HttpStatus.OK, description: 'transactions fetched successfully', type: [Transaction] })
+    @Get("/:adminId/transactions/providers/:providerId")
+    // get all transactions of a particular provider
+    async getProviderTransactions(
+        @Param("adminId", ParseIntPipe) adminId: number,
+        @Param("providerId", ParseIntPipe) providerId: number
+    ) {
+        // check admin
+        await this.adminService.getAdminWallet(adminId);
```

Comment:
implement all business logic to the service file, In the controller, you need to call just one method from the service.

do the same for all controllers